core/client/fs/src/main/java/alluxio/client/file/cache/store/RocksPageStore.java

```diff
@@ -180,25 +147,29 @@ private static PageId getPageIdFromKey(byte[] key) {
   }
 
   @Override
-  public long pages() {
-    return mSize.get();
-  }
-
-  @Override
-  public long bytes() {
-    return mBytes.get();
-  }
-
-  @Override
-  public Collection<PageInfo> getPages() {
-    try (RocksIterator iter = mDb.newIterator()) {
-      return Streams.stream(new PageIterator(iter)).collect(Collectors.toList());
+  public Stream<PageInfo> getPages() throws IOException {
+    try {
+      byte[] confData = mDb.get(CONF_KEY);
+      Cache.PRocksPageStoreOptions pOptions = mOptions.toProto();
+      if (confData != null) {
+        Cache.PRocksPageStoreOptions persistedOptions =
+            Cache.PRocksPageStoreOptions.parseFrom(confData);
+        if (!persistedOptions.equals(pOptions)) {
+          mDb.close();
```

Comment:
This is a pretty strong side effect from a function that looks like a getter. Shall we either rename this function or make it protected and close the db upon exception from `PageStore.create()`?

Reply:
make sense. Moved the conf verification part to factory method